### PR TITLE
Support for fit_features_together for implicit als

### PR DESCRIPTION
**Pull request by blondered**
- Added support for `fit_features_together`=True for implicit ALS model wrapper

### CODE REVIEW

File: rectools/models/implicit_als.py [hunk 1]
```diff
@@ -218,8 +212,237 @@ def _fit_paired_factors(
...
+            }
+        )
         features_model = CPUAlternatingLeastSquares(**features_model_params)
         features_model.item_factors = y_factors
         features_model.fit(xy_csr)
         x_factors = features_model.user_factors
     return x_factors
+
+
+def _init_latent_factors_cpu(
```
Comment (feldlime) [inline rectools/models/implicit_als.py:229]: Please add to this and to the following function short comments that the logic is copied and pasted from the implicit

File: tests/models/test_implicit_als.py [hunk 2]
```diff
@@ -105,6 +105,38 @@ def test_basic(
             actual,
         )
 
+    @pytest.mark.parametrize("fit_features_together", (False, True))
+    def test_consistent_with_pure_implicit(self, dataset: Dataset, fit_features_together: bool, use_gpu: bool) -> None:
+        base_model = AlternatingLeastSquares(factors=10, num_threads=2, iterations=30, use_gpu=use_gpu)
+        users = np.array([10, 20, 30, 40])
+        self._init_model_factors_inplace(base_model, dataset)
```
Comment (feldlime) [inline tests/models/test_implicit_als.py:112]: looks like a hack

Why not just fix `random_state`?

File: tests/models/test_implicit_als.py [hunk 3]
```diff
@@ -105,6 +105,38 @@ def test_basic(
...
+        actual_reco = wrapped_model.recommend(
+            users=users,
+            dataset=dataset,
+            k=3,
+            filter_viewed=False,
+        )
+
+        ui_csr = dataset.get_user_item_matrix(include_weights=True)
+        base_model.fit(ui_csr)
+        for user_id in users:
```
Comment (feldlime) [inline tests/models/test_implicit_als.py:126]: Isn't it correct that implicit can recommend for multiple users? Why do you need the cycle?

Comment (blondered) [reply tests/models/test_implicit_als.py:126]: I believe it's just simpler to match recos from different methods this way

File: rectools/models/implicit_als.py (hunk 1 above)
Comment (blondered) [reply rectools/models/implicit_als.py:229]: Done

**Review by feldlime — APPROVED** [review]

**Pull request merged**